extensions/wikia/GlobalWatchlist/GlobalWatchlistTask.class.php

```diff
@@ -43,34 +43,50 @@ public function addWatchers( $databaseKey, $nameSpace, array $watchers ) {
 
 			// Skip revisions that doesn't exist
 			if ( !empty( $revision ) ) {
+				$watchersToAdd = [];
 				$globalWatchlistBot = new GlobalWatchlistBot();
-
-				$db = wfGetDB( DB_MASTER, [ ], \F::app()->wg->ExternalDatawareDB );
 				foreach ( $watchers as $watcherID ) {
 					if ( $globalWatchlistBot->shouldNotSendDigest( $watcherID ) ) {
-						$this->clearGlobalWatchlistAll( $watcherID );
 						continue;
 					}
-
-					( new WikiaSQL() )
-						->INSERT()->INTO( GlobalWatchlistTable::TABLE_NAME )
-						->SET( GlobalWatchlistTable::COLUMN_USER_ID, $watcherID )
-						->SET( GlobalWatchlistTable::COLUMN_CITY_ID, \F::app()->wg->CityId )
-						->SET( GlobalWatchlistTable::COLUMN_TITLE, $databaseKey )
-						->SET( GlobalWatchlistTable::COLUMN_NAMESPACE, $nameSpace )
-						->SET( GlobalWatchlistTable::COLUMN_REVISION_ID, $revision->getId() )
-						->SET( GlobalWatchlistTable::COLUMN_REVISION_TIMESTAMP, $revision->getTimestamp() )
-						->SET( GlobalWatchlistTable::COLUMN_TIMESTAMP, $revision->getTimestamp() )
-						// Do nothing on duplicate key - we already have that record in place
-						->ON_DUPLICATE_KEY_UPDATE(
-							[ GlobalWatchlistTable::COLUMN_USER_ID => $watcherID ]
-						)
-						->run( $db );
+					$watchersToAdd[] = [
+						$watcherID,
+						\F::app()->wg->CityId,
+						$databaseKey,
+						$nameSpace,
+						$revision->getId(),
+						$revision->getTimestamp(),
+						$revision->getTimestamp()
+					];
 				}
+
+				$this->addWatchersToDb( $watchersToAdd );
```

Comment:
That's a good idea. I tried to check the flower logs to see how many users are actually being sent over during these giant tasks, but they've expired so I don't know. 1k seems like a sane default though. I'll update.